src/BLE_Server.cpp

```diff
@@ -142,13 +142,13 @@ void computeERG(int currentWatts, int setPoint) {
 
   if (cad > 20) {
     // 30  is amount of watts per shift. Was 50, seemed like too much...
-    if (abs(currentWatts - setPoint) < 30) {
-      amountToChangeIncline = (currentWatts - setPoint) * 1;
+    if (abs(currentWatts - setPoint) < WATTS_PER_SHIFT) {
+      amountToChangeIncline = (currentWatts - setPoint) * userConfig.getERGSensitivity();
```

Comment:
The ultimate change in `Main.cpp`'s `moveStepper()` function is still scaled by incline sensitivity. Should there be a flag that `moveStepper` can see to ignore inclineSensitivity in *erg mode* it it's only affected by the erg sensitivity setting?

Reply:
I think ultimately I'd like to get rid of inclineMultiplier entirely. I believe its redundant with a properly configured shiftStep

Reply:
Just playing around, felt like it was clear to have ERG compute as multiples of a shiftStep:
```
float amountToShift = (targetWatts - currentWatt) / WATTS_PER_SHIFT * userConfig.getERGSensitivity();
if (abs(amountToShift) > userConfig.getERGMaxShift()) {
  amountToShift = amountToShift / abs(amountToShift) * userConfig.getERGMaxShift();
}
userConfig.setIncline(incline + amountToShift * WATTS_PER_SHIFT);
```

## Assuming
```
WPS=30 ERG_SENS=0.9 ERG_MAXS=1.0 
```

## Example 1
```
CW=100 TW=200 
WC = TW-CW => 100
A2S = WC / WPS ==> 3.33
A2S = A2S * ERG_SENS ==> 2.99
IF ABS(A2S) > 1 THEN A2S = 1 ==> A2S = A2S / ABS(A2S) ==> 1
INCLINE = INCLINE + A2S * WPS ==> 30
```

## Example 2
```
CW=200 TW=100 
WC = TW-CW => -100
A2S = WC / WPS ==> -3.33
A2S = A2S * ERG_SENS ==> -2.99
IF ABS(A2S) > 1 THEN A2S = 1 ==> A2S = A2S / ABS(A2S) ==> -1
INCLINE = INCLINE + A2S * WPS ==> -30
```

## Example 3
```
CW=100 TW=125 
WC = TW-CW => 25
A2S = WC / WPS ==> 0.83
A2S = A2S * ERG_SENS ==> 0.75
IF ABS(A2S) > 1 THEN A2S = 1 ==> A2S = A2S / ABS(A2S) ==> 0.75
INCLINE = INCLINE + A2S * WPS ==> 22.5
```

## Example 4
```
CW=125 TW=100 
WC = TW-CW => -25
A2S = WC / WPS ==> -0.83
A2S = A2S * ERG_SENS ==> -0.75
IF ABS(A2S) > 1 THEN A2S = 1 ==> A2S = A2S / ABS(A2S) ==> -0.75
INCLINE = INCLINE + A2S * WPS ==> -22.5
```

Reply:
> Getting rid of it sounds good. It's a little confusing to have a setting both in the SS2K and in zwift related to incline difficulty.
> 
> Until then, what about something like this in `moveStepper`:
> 
> ```c
> void moveStepper(void *pvParameters) {
>   int acceleration   = maxStepperSpeed;
>   int targetPosition = 0;
>   int ustepChange = 0; 
> 
>   while (1) {
>     ustepChange = userConfig.getIncline();
>     if (!userConfig.getERGMode()) {
>       // Only use the incline multiplier when not in erg mode
>       ustepChange *= userConfig.getInclineMultiplier());
>     }
>     targetPosition = shifterPosition + ustepChange;
>     ...
> ```
> 
> It's a small change that's easy to get rid of if incline multiplier goes away. The flag already exists so not much effort there. And with this incline multiplier and erg sensitivity are untangled from each other.

Thinking out loud.......

I think some of the variable naming is throwing both you and @kadaan off, and I don't blame you. As the project has evolved, variable usage has subtly changed and possibly outgrown their intended purpose. This project needs more input from users like you guys to help become more accessible to others to modify. 

Maybe it's a naming issue, but (the value of) inclineMultiplier still needs to be incorporated into both modes to fully convert incline units to motor steps whether that incline is internally or externally derived.

To try to help, userConfig.getIncline() returns a target which is set either sim mode or internally via our ERG calc. Our ERG is purely specifying a change to incline to achieve target watts.

If you took into account my bike's normal settings (shiftStep = 800, inclineMultiplier=4), it would appear that (the value of) inclineMultiplier (4) could be derived (at least in my case) by doing (.005*userConfig.getShiftStep()). I think it's reasonable to assume .005 may be some type of constant that we can use to convert incline to motor steps. I'm also fairly certain that the constant of .005 is applicable across the board to all bikes. 

So we can get rid of inclineMultipler exactly that easy. Multiply userConfig.getShiftStep() by .005 . The reason I haven't is because apps other than Zwift may not have a "difficulty slider". The the inclineMultiplier is exactly a difficulty slider in sim mode. 

So we need a difficulty slider in some apps and not others but using this value currently affects our ERG mode (which is undesirable.) I suggest this option:

* Move inclineMultiplier's modification out of the motor loop and into bleServer.cpp's  onWrite() callback for sim mode where we have it modify userConfig.setIncline() directly. 
* Rename inclineMultiplier to "simModeDifficulty"
* Change motor control loop to convert userConfig.getIncline() into steps by doing: 
``const float inclineToStepConversion = userConfig.getShiftSteps() * .005;
targetPosition = shifterPosition + (userConfig.getIncline() * inclineToStepConversion);    ``

Therefore, it really doesn't matter on the stepper control loop which mode we're in. It's just going to convert userConfig.getIncline() to stepper steps and determine if that's a ccw or cw rotation.

Reply:
Modifying this however is outside of the scope of this particular PR, and i'll do it on a separate branch.

Reply:
Changing `inclineMultiplier` to `simModeDifficulty` is confusing as it is applicable to bother `Sim` and `ERG` modes.

Reply:
Regardless of the ultimate ERG/Sim algorithm, we'll probably need a set of unittests that can effectively do what @kadaan did w/ his examples above for different step sizes to make sure the microsteps scale properly as the algorithm gets tuned.

For that you'd want a slight refactor of `computeERG` so it returns the incline value rather than setting it directly. Might need some refactoring of `moveStepper` functionality to make it more unittest-able as well depending on how the algorithm is split between `computeERG`, `computeCSC` and `moveStepper`